test/functional/JIT_Test/src/jit/test/jitserver/JITServerTest.java

```diff
@@ -490,4 +594,78 @@ public void testServerComesUpAfterClientAndGoesDownAgain() throws IOException, I
 		logger.info("Stopping client...");
 		destroyAndCheckProcess(client, clientBuilder);
 	}
+
+	public void testServerSCC() throws IOException, InterruptedException {
+                logger.info("running testServerSCC: INFO and above level logging enabled");
+
+		// Run this tests only for the test variation with AOT Cache option specified
+		if(System.getProperty("CLIENT_PROGRAM").contains(aotCacheOption)) {
+			redirectProcessOutputs(clientBuilder, "testServerSCC.client");
+			redirectProcessOutputs(serverBuilder, "testServerSCC.server");
+
+			updateJITServerPort();
+
+			final Process server = startProcess(serverBuilder, "server");
+
+			Thread.sleep(SERVER_START_WAIT_TIME_MS);
+
+			Process client = startProcess(clientBuilder, "client");
+
+			logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			logger.info("Check whether the cache test_jitscc exists.");
+			if(checkCacheExists("test_jitscc"))
+				logger.info("Check if AOT methods exists in the cache test_jitscc::" + checkAOTCompileExists("test_jitscc"));
+			else
+				AssertJUnit.fail("The cache test_jitscc is not created.");
+
+			logger.info("Stopping client...");
+			destroyAndCheckProcess(client, clientBuilder);
+
+			logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			// Destroy the SCC and then restart the client
+			logger.info("Destroy the cache test_jitscc");
+			destroyCache("test_jitscc");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			// Check if the SCC exists and then restart the client
+			logger.info("Check whether the cache test_jitscc exists.");
+			if (checkCacheExists("test_jitscc"))
+				AssertJUnit.fail("The cache test_jitscc should have been destroyed.");
+
+			logger.info("Restarting client...");
+			client = startProcess(clientBuilder, "client");
+
+			logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			logger.info("Stopping client...");
+			destroyAndCheckProcess(client, clientBuilder);
+
+			logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			// Destroy the SCC and then restart the client
+			logger.info("Destroy the cache test_jitscc");
+			destroyCache("test_jitscc");
+			Thread.sleep(CLIENT_TEST_TIME_MS);
+
+			logger.info("Restarting client...");
```

Comment:
Why do we run the client 3 times? Twice should be enough. The first instance will populated the AOT cache at the server and the second instance will use AOT from the server.

Reply:
Also, make sure you destroy the SCC after you terminate the client. You cannot destroy a SCC if a JVM is attached to it.

Reply:
This code is removed.